### DISCUSSION TITLE
Share your winprops!

### DISCUSSION THREAD

**Discussion by develop7** [paperwm/PaperWM#971] [Show and tell]
Usually PaperWM does handle modal windows of all kinds quite well, but sometimes there are windows that aren't labeled as such but are modal in nature, i.e. Firefox's "About" window or Telegram's "Media viewer". Luckily, winprops settings and a "open on scratch layer" checkbox are to the rescue!

there are some of mine for the start:

| `title` | `wm_class` | scratch| Purpose |
|--------|--------|--------|--------|
| `Media viewer` | `/(org\.telegram\.desktop\|TelegramDesktop)/` | :heavy_check_mark:  | Prevents stacking Telegram's media viewer window |
| `/About.*?Firefox/` | `/firefox/` | :heavy_check_mark: | Prevents stacking Firefox's "About" window |
| N/A | `re.sonny.Junction` | :heavy_check_mark: | Prevents stacking of [Junction's](https://apps.gnome.org/Junction/) windows (TODO: limit it with a browser selector window) |

**Comment by Lythenas** [comment]
Awesome idea! Unfortunately I have little to share^^

| `title` | `wm_class` | scratch| Purpose |
|--------|--------|--------|--------|
| `Special Offers` | `steam` | :heavy_check_mark:  | Prevents stacking Steams startup popup window |

And rules for a couple of games matched by title to put the windows into the scratch layer. If they are otherwise really broken. But depends on the settings of the game, so I don't want to just list them.

**Comment by Thesola10** [comment]
Here's a few:

`title` | `wm_class` | scratch | Purpose
--------|------------|---------|---------
N/A     | `/^waydroid/i` | :heavy_check_mark: | Displays Waydroid apps properly
`wl-clipboard` | N/A | :heavy_check_mark: | Lets `wl-clipboard` launch in the background
`/Presentation:/` | `libreoffice-impress` | :heavy_check_mark: | Allows presentation UI to open on external displays (Needs localization per language)
N/A     | `ibus-extension-gtk3` | :heavy_check_mark: | Displays IBus suggestions popup properly
N/A     | `org.gnome.NautilusPreviewer` | :heavy_check_mark: | Prevents tiling Nautilus file preview popup
`Picture-in-Picture` | `/firefox/` | :heavy_check_mark: | Prevents tiling the Picture-in-Picture video player in Firefox (may need to be updated often, Mozilla changes the title every once in a while)

**Comment by rnevius** [comment]
Also nothing exciting to share, but since I often feel like I'm one of very few people using Zoom Workplace and PaperWM:

| `title` | `wm_class` | scratch | Purpose |
|---------|------------|---------|---------|
| N/A     | `zoom`     | ✔️      | Prevents tiling Zoom windows. Especially necessary for screen sharing controls. |

**Comment by caph1993** [comment]
Config for [Albert launcher](https://albertlauncher.github.io/)

| `title` | `wm_class` | scratch | Purpose |
|---------|------------|---------|---------|
| N/A     | `albert`     | ✔️      | Scratch is how launchers supposed to work. |
